SignalShareExtension/ShareViewController.swift

```diff
@@ -100,7 +185,44 @@ public class ShareViewController: UINavigationController, SAELoadViewDelegate {
 
         Logger.debug("\(self.logTag()) \(#function)")
 
-        // TODO:
+        if TSAccountManager.isRegistered() {
+            Logger.info("\(self.logTag) localNumber: \(TSAccountManager.localNumber)")
+
+            // We don't need to use ExperienceUpgradeFinder in the SAE.
+
+            // We don't need to use OWSDisappearingMessagesJob in the SAE.
+
+            OWSProfileManager.shared().ensureLocalProfileCached()
+        }
+    }
+
+    private func ensureRootViewController() {
+        Logger.debug("\(self.logTag()) \(#function)")
+
+        guard !TSDatabaseView.hasPendingViewRegistrations() else {
+            return
+        }
+        guard !hasInitialRootViewController else {
+            return
+        }
+        hasInitialRootViewController = true
+
+        Logger.info("Presenting initial root view controller")
+
+        if TSAccountManager.isRegistered() {
+            //                HomeViewController *homeView = [HomeViewController new];
+            //                SignalsNavigationController *navigationController =
+            //                    [[SignalsNavigationController alloc] initWithRootViewController:homeView];
+            //                self.window.rootViewController = navigationController;
+        } else {
```

Comment:
Here are the two hooks to integrate on.  I'm carrying the card in WIP to do the "please open main app to register" view.

Reply:
TIOLI, but I think quick and dirty is appropriate here. e.g. a simple alert.

Reply:
And leaving the loading view in place?  Hmm.